[HTTP/OAS] Add descriptions for Logstash pipeline APIs

lcawl opened the pull request:
## Summary

Relates to https://github.com/elastic/kibana/issues/180056

This PR adds operation summaries for the [Logstash configuration management APIs](https://www.elastic.co/guide/en/kibana/current/logstash-configuration-management-api.html) in the generated OpenAPI document.

NOTE: I've omitted the "access: public" from the "delete pipelines" and "get all pipelines" operations, since those don't currently exist in the documentation. I believe this means they'll use the default "internal" access value.

### How to test

Per https://github.com/elastic/kibana/pull/181277:

1. Start Elasticsearch (e.g. `yarn es snapshot --license trial`)
2. Add `server.oas.enabled: true` to `kibana.dev.yml`
3. Start Kibana (e.g. `yarn start --no-base-path`)
4. curl -s -u elastic http://localhost:5601/api/oas\?pathStartsWith\=/api/logstash/pipeline | jq > test.json

The output will contain info like this:

```
   "paths": {
    "/api/logstash/pipeline/{id}": {
      "delete": {
        "summary": "Delete a Logstash pipeline",
```

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `run` `docs-build` : Re-trigger the docs validation. (use unformatted text in the comment!)
</p>
</details>

karenzone commented on x-pack/plugins/logstash/server/routes/pipeline/delete.ts:
```diff
@@ -14,6 +14,10 @@ export function registerPipelineDeleteRoute(router: LogstashPluginRouter) {
       path: '/api/logstash/pipeline/{id}',
+      options: {
+        access: 'public',
+        description: `Delete a Logstash pipeline`,
```
```suggestion
        description: `Delete a centrally managed Logstash pipeline`,
```

karenzone replied:
```suggestion
        description: `Delete a centrally-managed Logstash pipeline`,
```

karenzone commented on x-pack/plugins/logstash/server/routes/pipeline/load.ts:
```diff
@@ -16,6 +16,10 @@ export function registerPipelineLoadRoute(router: LogstashPluginRouter) {
       path: '/api/logstash/pipeline/{id}',
+      options: {
+        access: 'public',
+        description: `Get a Logstash pipeline`,
```
```suggestion
        description: `Retrieve a centrally-managed Logstash pipeline`,
```

karenzone commented on x-pack/plugins/logstash/server/routes/pipeline/save.ts:
```diff
@@ -21,6 +21,10 @@ export function registerPipelineSaveRoute(
       path: '/api/logstash/pipeline/{id}',
+      options: {
+        access: 'public',
+        description: `Create a Logstash pipeline`,
```
```suggestion
        description: `Create a centrally-managed Logstash pipeline, or update existing`,
```

karenzone commented on x-pack/plugins/logstash/server/routes/pipelines/list.ts:
```diff
@@ -27,6 +27,10 @@ export function registerPipelinesListRoute(router: LogstashPluginRouter) {
       path: '/api/logstash/pipelines',
+      options: {
+        access: 'public',
+        description: `Get all Logstash pipelines`,
```
```suggestion
        description: `List all centrally-managed Logstash pipelines`,
```

karenzone reviewed (commented):
I left some suggestions inline for consideration. Do we have a character limit? The fact that these APIs work only for centrally-managed Logstash pipelines is significant.

lcawl commented:
> Do we have a character limit? The fact that these APIs work only for centrally-managed Logstash pipelines is significant.

Yes, this is the "operation summary" which is used as the title in the navigation bar, so we've added a linting rule to limit its length. Originally we were linting for strings to be less than 30, though I've updated that now to go as much as 45.

For the longer-form descriptions of the APIs (i.e. that exist in the body of the page, not the navigation tree), there's an outstanding issue to add support for that: https://github.com/elastic/kibana/issues/182779

karenzone commented:
Thanks for explaining the constraints. I modified my comments to use standard terminology and to respect the character limit... assuming that only display text counts against character limits.  :-)  Please LMKWYT

karenzone reviewed (approved):
LGTM (pending passing ci). :-)  Thanks for your work on this, @lcawl
